miqa/core/rest/scan.py

```diff
@@ -17,11 +19,31 @@ class Meta:
     decision = serializers.ChoiceField(choices=Scan.decision.field.choices)
 
 
+class DecisionSerializer(serializers.Serializer):
+    decision = serializers.ChoiceField(choices=Scan.decision.field.choices)
+
+
 class ScanViewSet(ReadOnlyModelViewSet):
     queryset = Scan.objects.all()
 
     filter_backends = [filters.DjangoFilterBackend]
     filterset_fields = ['experiment', 'site']
 
     permission_classes = [AllowAny]
-    serializer_class = ScanSerializer
+
+    def get_serializer_class(self):
```

Comment:
We can still use `ScanSerializer` as the default and override the serializer for specific actions. That may be a good call if we care about swagger UI working properly.

Reply:
[Here's](https://github.com/girder/dkc-next/blob/916c511f7ee0cb848e4dd5749ee4bd6e5d549165/dkc/core/rest/authorized_upload.py#L59) an example of doing that. I think you can also pass `serializer_class` to the action decorator itself, that may be sufficient, but I can't remember if I'm making that up or not.

Reply:
The only things you can pass to `@action` end in `_classes` (i.e. `permissions_classes`), so I don't think you can set the serializer.
This way you can use `self.get_serializer` rather than specify which serializer using in the `@swagger_auto_schema` and explicitly when you use it, but I can do it that way if you think it's cleaner.